src/app/js/formats/istexSummary/FetchFold.js

```diff
@@ -0,0 +1,129 @@
+        marginLeft: 8,
+    },
+    arrowClose: {
+        transform: 'rotate(-90deg)',
+    },
+});
+
+const circularProgress = (
+    <CircularProgress
+        size={20}
+        innerStyle={{
+            display: 'flex',
+            marginLeft: 8,
+        }}
+    />
+);
+
+export class FetchFold extends Component {
+    state = {
+        data: null,
+        error: null,
+        isLoading: false,
+        isOpen: false,
+    };
+
+    open = () => {
+        if (this.state.data) {
+            this.setState({
+                isOpen: true,
+            });
+            return;
+        }
+
+        this.setState({ isLoading: true }, () => {
+            this.props
+                .getData(this.props)
+                .then(data => {
+                    this.setState({
+                        data,
+                        isLoading: false,
+                        isOpen: true,
+                    });
+                })
+                .catch(error => {
+                    console.error(error);
+                    this.setState({ error: true });
+                });
+        });
+    };
+
+    close = () => {
+        this.setState({ isOpen: false });
+    };
+
+    render() {
+        const { label, p: polyglot, renderData } = this.props;
+        const { error, data, isOpen, isLoading } = this.state;
+
+        if (error) {
+            return <Alert>{polyglot.t('istex_error')}</Alert>;
+        }
+
+        return (
+            <div className="istex-fold">
+                <div>
+                    <Button onClick={isOpen ? this.close : this.open}>
+                        <div className={css(styles.buttonLabel)}>
+                            <Arrow
+                                className={
+                                    isOpen ? undefined : css(styles.arrowClose)
+                                }
+                            />
+                            {isOpen ? <FolderOpen /> : <Folder />}
+                            <span className={css(styles.labelText)}>
+                                {label}
+                            </span>
+                            {isLoading && circularProgress}
+                        </div>
+                    </Button>
+                    {isOpen && (
+                        <ul>
+                            {data.map(value => (
+                                <li
+                                    key={get(value, 'id', value)}
+                                    className={css(styles.li)}
+                                >
+                                    {renderData(value)}
+                                </li>
+                            ))}
+                        </ul>
+                    )}
+                </div>
+            </div>
+        );
+    }
+}
+
+FetchFold.propTypes = {
+    label: PropTypes.string.isRequired,
+    p: polyglotPropTypes.isRequired,
+    getData: PropTypes.func.isRequired,
+    renderData: PropTypes.func.isRequired,
```

Comment:
It would be far better if you use a render prop as `children()` instead of renderData